### PR TITLE
Add option to ignore some image regions during comparison

### PR DESCRIPTION
**Pull request by pashidlos**
Sometimes image may contain dynamic data or information that is not important in current test
With the help of `canvajs` you could draw rectangle to ignore them 
Before comparing image with baseline we clear all pixels in ignored regions and diff image will have them in white 

Quick demo: https://monosnap.com/file/T6xAORxOuhAaAWfcLNFFU7DF5hGxW6

### CODE REVIEW

**Comment by MitkoTschimev** [comment]
this is already build in ;) 
https://doc.visual-knight.io/adapters/codeceptjs -> 
```typescript
/**
 * @param testName {string} Is provided to visual knight (must be unique)
 * @param options {ScreenshotOptions} Contains additional settings
 */
I.compareFullpageScreenshot(testName, options)
```

the options parameter

```
/*
ScreenshotOptions {
  hide?: string[] // Array of css selectors which gets hidden by "opacity: 0",
  remove?: string[] // Array of css selectors which gets hidden by "display: none",
  additional?: object // Data is saved as relation to the variation. (Future: can be used for filtering)
}
*/
```

**Comment by pashidlos** [comment]
I know this option but for me css is not enougth and they are only applicable for web applications

**Comment by MitkoTschimev** [comment]
what do you mean exactly :)
You mean if you are testing mobile apps? iOS or Android?

File: apps/api/src/app/comparison/services/comparison.service.ts [hunk 1]
```diff
@@ -364,11 +375,26 @@ export class ComparisonService {
...
-        state,
+        state
       }
     });
 
     console.log('Update image data');
     return from(testSession.then());
   }
+
+  private applyIgnoreAreas(image: PNG, ignoreAreas: IgnoreAreaType[]): Buffer {
```
Comment (MitkoTschimev) [inline apps/api/src/app/comparison/services/comparison.service.ts:386]: how fast is this method?

**Comment by MitkoTschimev** [comment]
I like the idea for non web apps and we should add it as additional feature

File: apps/api/src/app/comparison/services/comparison.service.ts (hunk 1 above)
Comment (pashidlos) [reply apps/api/src/app/comparison/services/comparison.service.ts:386]: not sure that I'm able to answer that question directly 
as `pixelmatch` doesn't have this functionality they propose this as workaround https://github.com/mapbox/pixelmatch/issues/9#issuecomment-499665722
it completely depends on image size 
do you know how we could test it?

**Pull request merged**